dcc-submission-loader/src/main/java/org/icgc/dcc/submission/loader/core/DependencyFactory.java

```diff
@@ -0,0 +1,215 @@
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.submission.loader.core;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static lombok.AccessLevel.PRIVATE;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.val;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.compress.CompressionCodecFactory;
+import org.icgc.dcc.common.core.meta.Resolver.DictionaryResolver;
+import org.icgc.dcc.common.core.meta.RestfulDictionaryResolver;
+import org.icgc.dcc.submission.loader.cli.ClientOptions;
+import org.icgc.dcc.submission.loader.db.orientdb.OrientdbDatabseService;
+import org.icgc.dcc.submission.loader.db.orientdb.OrientdbDocumentLinker;
+import org.icgc.dcc.submission.loader.db.postgres.PostgresDatabaseService;
+import org.icgc.dcc.submission.loader.io.LoadFilesResolver;
+import org.icgc.dcc.submission.loader.meta.ReleaseResolver;
+import org.icgc.dcc.submission.loader.meta.SubmissionMetadataService;
+import org.icgc.dcc.submission.loader.meta.TypeDefGraph;
+import org.postgresql.ds.PGPoolingDataSource;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import com.orientechnologies.orient.core.db.OPartitionedDatabasePool;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+
+@RequiredArgsConstructor(access = PRIVATE)
+public final class DependencyFactory implements Closeable {
```

Comment:
I think we should switch to Spring now that this project is becoming larger and appears to have some momentum behind it / is quite valuable.